### PR TITLE
Decouple swift from apple platform

### PR DESCRIPTION
**Pull request by nguyentruongtho**
Code base for supporting swift is getting larger, we should think about how to make buck to build swift platform-independently. This PR is a starting point for that, it is extracted from https://github.com/facebook/buck/pull/893, and should make it easier to specify swift version to be used with buck.

### CODE REVIEW

**Comment by facebook-github-bot-3** [comment]
By analyzing the blame information on this pull request, we identified @Coneko, @yiding and @ryu2 to be potential reviewers.

**Comment by facebook-github-bot-8** [comment]
@nguyentruongtho updated the pull request - [view changes](https://github.com/facebook/buck/pull/896/files/b936ee5ba4cfcbc53bb1e56de63f6d1e1c79e877..20e8a366b21eb78c29d73a9f7b7209f416b09529)

**Comment by facebook-github-bot-7** [comment]
Thanks for importing. If you are a Facebook employee, you can view this diff [on Phabricator](https://phabricator.intern.facebook.com/D3918972).

File: test/com/facebook/buck/swift/SwiftDescriptionsTest.java [hunk 1]
```diff
@@ -57,13 +57,13 @@ public void testPopulateSwiftLibraryDescriptionArg() throws Exception {
...
     SwiftDescriptions.populateSwiftLibraryDescriptionArg(pathResolver, output, args, buildTarget);
-    assertThat(output.moduleName.get(), Matchers.equalTo("bar"));
-    assertThat(output.srcs.get(), Matchers.equalTo(ImmutableSortedSet.<SourcePath>of(swiftSrc)));
+    assertEquals(output.moduleName.get(), "bar");
+    assertEquals(output.srcs.get(), ImmutableSortedSet.<SourcePath>of(swiftSrc));
 
     args.moduleName = Optional.of("baz");
 
     SwiftDescriptions.populateSwiftLibraryDescriptionArg(pathResolver, output, args, buildTarget);
-    assertThat(output.moduleName.get(), Matchers.equalTo("baz"));
```
Comment (Coneko) [inline test/com/facebook/buck/swift/SwiftDescriptionsTest.java]: We use `assertThat` as much as possible so that the failure print outs are consistent.

Comment (Coneko) [inline test/com/facebook/buck/swift/SwiftDescriptionsTest.java]: Also note that changing them to `assertEquals` without flipping the arguments is wrong: `assertEquals`'s order is expected, actual. That's quite unintuitive and another reason we use `assertThat`.

Comment (nguyentruongtho) [inline test/com/facebook/buck/swift/SwiftDescriptionsTest.java]: 👍  It was a quick fix for `lint` issues, I prefer using `assertThat` also.

File: src/com/facebook/buck/swift/BUCK [hunk 2]
```diff
@@ -1,4 +1,19 @@
...
+    'SwiftPlatform',
+  ],
+  autodeps = True,
+  visibility = ['PUBLIC'],
+)
+
 standard_java_library(
+  srcs = glob(['*.java',], excludes=PLATFORM_SRCS),
   tests = [
     '//test/com/facebook/buck/swift:swift',
```
Comment (mkillianey) [inline src/com/facebook/buck/swift/BUCK]: In `test/com/facebook/buck/swift/BUCK`, this commit removed the swift:swift target and replaces it with `unit` and `swift_integration_test`, so this no longer points to a valid target.

**Pull request closed without merging**